solutions/swb-app/src/costCenterRoutes.ts

```diff
@@ -35,7 +41,16 @@ export function setUpCostCenterRoutes(router: Router, costCenterService: CostCen
         DeleteCostCenterRequestParser,
         deleteCostCenterRequest
       );
-      res.status(204).send(await costCenterService.softDeleteCostCenter(validatedRequest));
+
+      async function checkDependency(costCenterId: string): Promise<void> {
```

Comment:
What's the point of passing the function to have it call from within the soft delete method? Why not just call this here?